docs/API.md

```diff
@@ -0,0 +1,203 @@
+    ```
+    
+*   Query - name\_nft\_address
+    *   Returns the address of the NFT contract
+    
+    ```plain
+    {endpoint}/cosmwasm/v1/contract/{resolver_contract_address}/smart/{base64Encoding({ "name_nft_address": { }})}
+    ```
+    
+*   Query - referral\_count
+    *   Returns the number of referrals by the given ICNS name
+    
+    ```plain
+    {endpoint}/cosmwasm/v1/contract/{resolver_contract_address}/smart/{base64Encoding({ "referral_count": { "name": {twitter_username} }})}
+    ```
+    
+*   Query - fee
+    *   Returns the current fee required for claiming/minting a new name
+    
+    ```plain
+    {endpoint}/cosmwasm/v1/contract/{resolver_contract_address}/smart/{base64Encoding({ "fee": { } })}
+    ```
+    
+*   Query - name\_by\_twitter\_id
+    *   Returns the name of the user by Twitter ID (note that Twitter ID different from Twitter handle) **at the time the name was claimed.**
+    
+    ```plain
+    {endpoint}/cosmwasm/v1/contract/{resolver_contract_address}/smart/{base64Encoding({ "name_by_twitter_id": { "twitter_id": {twitter_id} }})}
+    ```
+    
+*   Execute Message - claim
+*   Execute Message - update\_verifier\_pub\_keys
+*   Execute Message - set\_verification\_threshold
+*   Execute Message - set\_name\_nft\_address
+*   Execute Message - set\_minting\_fee
+*   Execute Message - withdraw\_funds
+
+### **Name NFT API**
+
+*   Query - admin
+    *   Returns the admin address
+*   Query - is\_admin
+    *   Checks whether the specified address is an admin
+*   Query - transferrable
+    *   Returns whether Name NFTs are currently transferrable
+*   Query - owner\_of
+    *   Check the owner of the given token ID (optional filter on including expired owners)
+*   Query - approval
+    *   Check for the approval response for a given token ID and spender (optional filter on including expired owners)
+*   Query - approvals
+    *   Check for the approval response for a given token ID (optional filter on including expired owners)
+*   Query - all\_operators
+*   Query - num\_tokens
+*   Query - contract\_info
+*   Query - nft\_info
+*   Query - all\_nft\_info
+*   Query - tokens
+*   Query - all\_tokens
+*   Query - minter
+
+Examples
+--------
+
+**Get a bech32 address from an ICNS name**
+
+```plain
+{endpoint}/cosmwasm/v1/contract/{resolver_contract_address}/smart/{base64Encoding({ "address-by-icns": { "icns": {twitter_username}.{bech32_prefix} }})}
+```
+
+*   Example: [https://lcd.testnet.osmosis.zone/cosmwasm/wasm/v1/contract/osmo1qjs29tsavxuk39dugs7lny7cc28a57hqa9u9q0ysr2qat96aghrqzgp60x/smart/eyJpY25zX25hbWVzIjogeyJhZGRyZXNzIjogIm9zbW8xdnY2aHJ1cXV6cHR5NHhwa3M5em5rdzhneXM1eDRuc25nNGtlcnkifX0K](https://lcd.testnet.osmosis.zone/cosmwasm/wasm/v1/contract/osmo1qjs29tsavxuk39dugs7lny7cc28a57hqa9u9q0ysr2qat96aghrqzgp60x/smart/eyJpY25zX25hbWVzIjogeyJhZGRyZXNzIjogIm9zbW8xdnY2aHJ1cXV6cHR5NHhwa3M5em5rdzhneXM1eDRuc25nNGtlcnkifX0K)
```

Comment:
```suggestion
*   Example: [https://lcd.osmosis.zone/cosmwasm/wasm/v1/contract/osmo1qjs29tsavxuk39dugs7lny7cc28a57hqa9u9q0ysr2qat96aghrqzgp60x/smart/eyJpY25zX25hbWVzIjogeyJhZGRyZXNzIjogIm9zbW8xdnY2aHJ1cXV6cHR5NHhwa3M5em5rdzhneXM1eDRuc25nNGtlcnkifX0K](https://lcd.osmosis.zone//cosmwasm/wasm/v1/contract/osmo1qjs29tsavxuk39dugs7lny7cc28a57hqa9u9q0ysr2qat96aghrqzgp60x/smart/eyJpY25zX25hbWVzIjogeyJhZGRyZXNzIjogIm9zbW8xdnY2aHJ1cXV6cHR5NHhwa3M5em5rdzhneXM1eDRuc25nNGtlcnkifX0K)
```

Reply:
In addition to the endpoint update to mainnet, the Contract address needs updated to mainnet version and the query was an `icns_names` one so updated with a `address_by_icns` one.
```suggestion
*   Example: [https://lcd.osmosis.zone/cosmwasm/wasm/v1/contract/osmo1xk0s8xgktn9x5vwcgtjdxqzadg88fgn33p8u9cnpdxwemvxscvast52cdd/smart/eyJhZGRyZXNzX2J5X2ljbnMiOiB7ImljbnMiOiAiTm9zTm9kZV8ub3NtbyJ9fQo=](https://lcd.osmosis.zone/cosmwasm/wasm/v1/contract/osmo1xk0s8xgktn9x5vwcgtjdxqzadg88fgn33p8u9cnpdxwemvxscvast52cdd/smart/eyJhZGRyZXNzX2J5X2ljbnMiOiB7ImljbnMiOiAiTm9zTm9kZV8ub3NtbyJ9fQo=)
```